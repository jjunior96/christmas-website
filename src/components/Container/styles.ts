import styled, { css } from 'styled-components';

export const Container = styled.div`
  display: flex;

  width: 100%;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.grid.container};

    display: flex;
    flex-direction: column;
  `}
`;
