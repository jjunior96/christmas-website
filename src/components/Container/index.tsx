import * as S from './styles';

interface ContainerProps {
  children: React.ReactNode;
}

const Container = ({ children }: ContainerProps) => (
  <S.Container>
    <S.Content>{children}</S.Content>
  </S.Container>
);

export default Container;
