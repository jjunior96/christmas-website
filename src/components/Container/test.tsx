import { renderWithTheme } from '../../utils/tests/helpers';

import Container from '.';

describe('<Container />', () => {
  it('should render the component', () => {
    const { container } = renderWithTheme(
      <Container>
        <h1>Next Example</h1>
      </Container>
    );

    expect(container).toMatchSnapshot();
  });
});
