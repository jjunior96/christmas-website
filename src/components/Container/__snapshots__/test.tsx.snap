// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render the component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  width: 100%;
  max-width: 130rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Next Example
      </h1>
    </div>
  </div>
</div>
`;
